File: Leetcode/1399. Count Largest Group.py
class Solution:
    def countLargestGroup(self, n: int) -> int:
        def sumOfDigits(num):
            s=0
            while num>0:
                s += num%10
                num //=10
            return s
        freq={}
        maxFrequency=-float('inf')
        for i in range(1,n+1):
            # t=(i-1)%9 #this only works if we want to reduce to one digit
            t = sumOfDigits(i)
            if t in freq:
                freq[t] = freq[t] + 1
            else:
                freq[t] = 1
            maxFrequency = max(maxFrequency,freq[t])
        res=0
        for i in freq:
            if freq[i] == maxFrequency:
                res+=1
        return res
        

File: Leetcode/1399. Count Largest Group.cpp
class Solution {
public:
    int sumOfDigits(int num){
        int s=0;
        while(num){
            s+=num%10;
            num/=10;
        }
        return s;
    }
    int countLargestGroup(int n) {
        unordered_map<int,int> m;
        int maxFrequency = INT_MIN;
        for(int i=1;i<=n;i++){
            int t=sumOfDigits(i);
            maxFrequency = max(++m[t],maxFrequency);
        }
        int res=0;
        for(auto i=m.begin();i!=m.end();i++){
            if(i->second == maxFrequency){
                res++;
            }
        }
        return res;
    }
};
